factory/processing/processComponentsHandler.go

```diff
@@ -593,7 +593,7 @@ func (m *managedProcessComponents) ESDTDataStorageHandlerForAPI() vmcommon.ESDTN
 	m.mutProcessComponents.RLock()
 	defer m.mutProcessComponents.RUnlock()
 
-	if m.esdtDataStorageForApi == nil {
+	if m.processComponents == nil {
```

Comment:
So the previous check is now useless, right?

Reply:
all the functions behave the same: guard `m.processComponents` for `nil` since further access would make the code panic. Anyway, returning `nil` or `m.processComponents.esdtDataStorageForApi` which is nil, both will end up returning `nil`. It was an older code issue that I've fixed now